src/Core/Scripting/OOJSPlayerShip.m

```diff
@@ -678,7 +679,41 @@ static JSBool PlayerShipSetProperty(JSContext *context, JSObject *this, jsid pro
 				return YES;
 			}
 			break;
-			
+		
+		case kPlayerShip_compassTarget:
+			// can't change compass target in basic mode
+			if (![player hasEquipmentItemProviding:@"EQ_ADVANCED_COMPASS"]) 
+			{
+				OOJSReportError(context, @"Compass target cannot be set with a basic compass.");
+				return NO;
+			}
+			// make sure we have a valid entity
+			if (!JSVAL_IS_NULL(*value) && JSValueToEntity(context, *value, &eValue)) 
+			{
+				Entity *current = [player compassTarget];
+				[player setNextCompassMode];
+				[player validateCompassTarget];
+				// cycle the targets until we either get back to the start (entity not found) or we find the one we're looking for
+				while ([player compassTarget] != current && [player compassTarget] != eValue)
+				{
+					[player setNextCompassMode];
+					[player validateCompassTarget];
+				}
+				if ([player compassTarget] == current)
+				{ 
+					return NO;
+				}
+				else
+				{
+					return YES;
+				}
```

Comment:
Lines 702 ti 706 can be simplified to 
`return [player compassTarget] != current;`